Refactor complexity of validation method using string methods extension

RamonMartinezNieto opened the pull request:
Refactor to reduce if concatenation.

It's a very simple thing and maybe unnecessary. I hate concatenating ifs.

Regards.

kayhantolga commented on OpenAI.SDK/OpenAiOptions.cs:
```diff
@@ -164,32 +166,20 @@ private static OpenAiOptions CreateAzureSettingsWithBaseDomain(string apiKey, st
-                throw new ArgumentNullException(nameof(ResourceName));
-            }
+            if (ProviderType == ProviderType.Azure)
+                ResourceName?.ThrowIfIsNullOrEmpty(nameof(ResourceName));
```
If ResourceName is null, it won't throw exception.

kayhantolga commented on OpenAI.SDK/OpenAiOptions.cs:
```diff
@@ -164,32 +166,20 @@ private static OpenAiOptions CreateAzureSettingsWithBaseDomain(string apiKey, st
-                throw new ArgumentNullException(nameof(DeploymentId));
-            }
+            DeploymentId?.ThrowIfIsNullOrEmpty(nameof(DeploymentId));
 
```
If DeploymentId is null here, it won't throw exception.

kayhantolga commented:
Hi Martinez,
Thank you for your contribution. I appreciate your effort and agree that simplifying code is a great idea. However, I noticed some bugs in your code. If you could address these issues, I would be happy to merge your pull request.

RamonMartinezNieto commented:
Added nullable value type to string (to avoid warnings) and fixed the two bugs.

Sorry for my mistakes and thank you for kindly pointing them out to me. Thank you for your work and effort in this repository, if there is anything I can help with, I'll be happy to do so.

kayhantolga commented:
Hi again, please try again with these settings. You will see your method will not be triggered when `deploymentId `is null
```
 "OpenAIServiceOptions": {
    "ApiKey": "Put your api key to here or to User Secret file", //How can I get an app key? visit here: https://openai.com/
   // "DeploymentId": "Your Deployment Id",
    "ResourceName": "",
    "ProviderType": "Azure"
    ,"ApiVersion": "Your API Version"//optional
  }
  ```
